tests/behaviour/block_data_test.py

```diff
@@ -0,0 +1,46 @@
+import os
+
+import pytest
+from starkware.starknet.compiler.compile import compile_starknet_files
+from starkware.starknet.testing.state import StarknetState
+
+from tests.logging.generateMarkdown import steps_in_function
+from tests.utils import CAIRO_PATH
+from warp.yul.main import transpile_from_solidity
+from warp.yul.starknet_utils import deploy_contract, invoke_method
+
+test_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+@pytest.mark.asyncio
+async def test_starknet():
+    block_data_sol = os.path.join(test_dir, "block_data.sol")
+    block_data_info = transpile_from_solidity(block_data_sol, "WARP")
+    block_data_cairo = os.path.join(test_dir, "block_data.cairo")
+
+    with open(block_data_cairo, "w") as f:
+        f.write(block_data_info["cairo_code"])
+
+    block_data_contractDef = compile_starknet_files(
+        [block_data_cairo], debug_info=True, cairo_path=[CAIRO_PATH]
+    )
+
+    starknet = await StarknetState.empty()
+    (block_data_address, _) = await deploy_contract(
+        starknet, block_data_info, block_data_contractDef
+    )
+
+    block_number = await invoke_method(
+        starknet, block_data_info, block_data_address, "getBlockNumber"
+    )
+    assert block_number.retdata == [
+        32,
+        2,
+        0,
+        3618502788666131213697322783095070105623107215331596699973092056135872020480,
```

Comment:
What is that number? How do we know it's going to be it?

Reply:
Not sure, if you run this on an empty starknet instance it gives you the same number:
```
%lang starknet
from starkware.starknet.common.syscalls import get_block_number, get_block_timestamp

@external
func testMe{syscall_ptr : felt*}() -> (res: felt):
    let (res_1) = get_block_number()
    return (res=res_1)
end
```